src/doc/book/lifetimes.md

```diff
@@ -56,6 +56,8 @@ To fix this, we have to make sure that step four never happens after step
 three. The ownership system in Rust does this through a concept called
 lifetimes, which describe the scope that a reference is valid for.
 
+**Note** It's important to understand that lifetime annotations are _descriptive_, not _prescriptive_. This means that how long a reference is valid  is determined by the code, not by the annotations. The annotations, however, point out this fact to the compiler in case it can't figure it out by itself.
```

Comment:
> The annotations, however, point out this fact to the compiler in case it can't figure it out by itself.

So, the compiler doesn't try to "figure out" the right lifetimes; it only applies the elision rules, which are just dumb patterns. So I would change this to something like,

> The annotations, however, give the compiler the information it needs for more complex cases.

Or something like that. What do you think?